app/controllers/my/profile_controller.rb

```diff
@@ -32,17 +32,45 @@ def create
   end
 
   def edit
-    render_modal("edit-profile", "edit")
+    @user = current_user
+
+    respond_to do |format|
+      format.js { render_modal("edit-profile", "_edit_modal") }
+      format.html { setup_solutions }
+    end
   end
 
   def update
-    @profile.update(profile_params)
-    User.find(current_user.id).update(user_params)
-    redirect_to @profile
+    @user = current_user
+
+    ActiveRecord::Base.transaction do
+      if @profile.update(profile_params) && @user.update(user_params)
+        redirect_to @profile
+      else
+        setup_solutions
+        render :edit
+      end
+    end
   end
 
   private
 
+  def setup_solutions
```

Comment:
Could we either
1) Move this somewhere to avoid the duplication across controllers
2) Redirect to `action: :show` with a query param that means the modal fires there instead?

Reply:
I don't think we can merge with the duplication because we'll totally forget to change something in one place next time we make a modification.